commands/utils/params.go

```diff
@@ -110,140 +110,219 @@ type Scan struct {
+	if pullRequestIDString := getTrimmedEnv(GitPullRequestIDEnv); pullRequestIDString != "" {
+		if g.PullRequestID, err = strconv.Atoi(pullRequestIDString); err != nil {
+			return err
+		}
+	}
+	return
+}
+
 func GetFrogbotUtils() (frogbotUtils *FrogbotUtils, err error) {
 	// Get server and git details
-	server, gitParams, err := extractEnvParams()
+	server, clientInfo, err := extractClientServerParams()
 	if err != nil {
 		return nil, err
 	}
 	defer func() {
-		e := SanitizeEnv()
-		if err == nil {
-			err = e
-		} else if e != nil {
-			err = fmt.Errorf("%s\n%s", err.Error(), e.Error())
-		}
+		err = errors.Join(err, SanitizeEnv())
 	}()
 
-	// Build Version control client for REST API requests
+	// Build a version control client for REST API requests
 	client, err := vcsclient.
-		NewClientBuilder(gitParams.GitProvider).
-		ApiEndpoint(gitParams.ApiEndpoint).
-		Token(gitParams.Token).
-		Project(gitParams.GitProject).
+		NewClientBuilder(clientInfo.GitProvider).
+		ApiEndpoint(clientInfo.APIEndpoint).
+		Token(clientInfo.Token).
+		Project(clientInfo.Project).
 		Logger(log.GetLogger()).
-		Username(gitParams.Username).
+		Username(clientInfo.Username).
 		Build()
 	if err != nil {
 		return nil, err
 	}
 
-	configAggregator, err := getConfigAggregator(client, server, gitParams)
-	return &FrogbotUtils{ConfigAggregator: configAggregator, Client: client, ServerDetails: server}, err
-}
-
-// getConfigAggregator returns a FrogbotConfigAggregator based on frogbot-config.yml and environment variables.
-func getConfigAggregator(client vcsclient.VcsClient, server *coreconfig.ServerDetails, gitParams *Git) (FrogbotConfigAggregator, error) {
-	releasesRepo, err := downloadExtractorsFromRemoteIfNeeded(server, "")
+	configAggregator, err := getConfigAggregator(client, clientInfo, server)
 	if err != nil {
 		return nil, err
 	}
-	configFileContent, err := getConfigFileContent(client)
-	// If there is a missing configuration file error, try to generate an environment variable-based config aggregator.
-	if _, missingConfigErr := err.(*ErrMissingConfig); missingConfigErr {
-		log.Debug("Unable to retrieve", FrogbotConfigFile, err.Error())
-		configAggregator, err := newConfigAggregatorFromEnv(gitParams, server, releasesRepo)
-		if err != nil {
-			return nil, err
-		}
-		return configAggregator, err
-	} else if err != nil {
+	return &FrogbotUtils{Repositories: configAggregator, Client: client, ServerDetails: server}, err
+}
+
+// getConfigAggregator returns a RepoAggregator based on frogbot-config.yml and environment variables.
+func getConfigAggregator(client vcsclient.VcsClient, clientInfo *ClientInfo, server *coreconfig.ServerDetails) (RepoAggregator, error) {
+	configFileContent, err := getConfigFileContent(client, clientInfo)
+	// Don't return error in case of a missing frogbot-config.yml file
+	// If an error occurs due to a missing file,
+	// Attempt to generate an environment variable-based configuration aggregator as an alternative.
```

Comment:
The above 3 lines include one sentence. correct? If so, then only the first word in the sentence should start with a capital letter.

Reply:
done